client/src/components/core/ms-modal/MsPasswordInputModal.vue

```diff
@@ -0,0 +1,76 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <ms-modal
+    :title="title"
+    :subtitle="subtitle"
+    :close-button="{ visible: true }"
+    :confirm-button="{
+      label: okButtonText || $t('PasswordInputModal.ok'),
+      disabled: password.length === 0,
+      onClick: confirm,
+    }"
+  >
+    <ms-password-input
+      :label="inputLabel || ''"
+      v-model="password"
+      @on-enter-keyup="confirm()"
+    />
+  </ms-modal>
+</template>
+
+<script lang="ts">
+import MsPasswordInputModal from '@/components/core/ms-modal/MsPasswordInputModal.vue';
+import { getClientInfo } from '@/parsec/login';
+import { ComposerTranslation } from 'vue-i18n';
+
+export interface GetPasswordOptions {
+  title: string;
+  subtitle?: string;
+  inputLabel?: string;
+  okButtonText?: string;
+}
+
+export async function getPasswordFromUser(t: ComposerTranslation): Promise<string | null> {
+  const clientInfo = await getClientInfo();
```

Comment:
Probably a bit of a misunderstanding here. `getPasswordFromUser` should take a `GetPasswordOptions`, not a `ComposerTranslation`. Title, subtitles and all should be decided by the caller. All I was trying to say was that you could do
```typescript
const modal = await modalController.create({
  ...
  componentProps: options,
});
```

instead of
```typescript
const modal = await modalController.create({
  ...
  componentProps: {
    title: options.title,
    subtitle: options.subtitle,
    ...
  }
});
```